backend/app/Http/Controllers/StackController.php

```diff
@@ -4,28 +4,48 @@
 
 use Exception;
 use App\Models\Stack;
+use Illuminate\Http\Request;
 use Illuminate\Http\JsonResponse;
 use App\Http\Requests\StackRequest;
+use Illuminate\Support\Str;
 use Symfony\Component\HttpFoundation\Response;
 
 class StackController extends Controller
 {
-    public function addStack(StackRequest $request): JsonResponse
+
+    public function addStack(Request $request)
     {
-        $data = $request->all();
-        try {
-            if($data){
-                Stack::create($data);
-                return $this->sendResponse(false, null, 'Stack created', $data, Response::HTTP_CREATED);
-            }else {
-                return $this->sendResponse(true, 'Stack failed', Response::HTTP_BAD_REQUEST);
+        try{
+            $payload = json_decode($request->getContent(), true);
+    
+            if(!isset($payload["name"])){
+                return $this->sendResponse(true, "expected a valid payload", "invalid payload given.", null, Response::HTTP_BAD_REQUEST);
+            }
+            
+            $id = Str::uuid();
+            $name = $payload["name"];
+            
+            $restStack = Stack::where("name", $name);
```

Comment:
what's this line doing when you are adding a new stack?

Reply:
to ensure no user is duplicating a stack in the database

Reply:
ok, but it can be done in a better manner